ui/src/components/DetectionsTable.tsx

```diff
@@ -16,18 +26,30 @@ import { DetectionsPlayer } from "./Player/DetectionsPlayer";
 export default function DetectionsTable({
   detections,
   feed,
-  candidate
+  candidate,
 }: {
   detections: Detection[];
   feed: Pick<Feed, "slug" | "nodeName">;
-  candidate: Pick<Candidate, "id">;
+  candidate: Pick<Candidate, "id" | "visible">;
 }) {
   const offsetPadding = 15;
   const minOffset = Math.min(...detections.map((d) => +d.playerOffset));
   const maxOffset = Math.max(...detections.map((d) => +d.playerOffset));
   const startOffset = Math.max(0, minOffset - offsetPadding);
   const endOffset = maxOffset + offsetPadding;
 
+  const { currentUser } = useGetCurrentUserQuery().data ?? {};
+
+  const queryClient = useQueryClient();
+
+  const setDetectionVisible = useSetDetectionVisibleMutation({
+    onSuccess: () => {
+      queryClient.invalidateQueries(
+        useCandidateQuery.getKey({ id: candidate.id }),
```

Comment:
This looks good but I just noticed that the invalidation doesn't automatically trigger an update to the UI (since there's nothing to trigger a re-render).

I think ideally we should do something like [this](https://tanstack.com/query/v4/docs/react/guides/updates-from-mutation-responses) or even better, do it [optimistically](https://tanstack.com/query/v4/docs/react/guides/optimistic-updates)

Reply:
Actually, looking at it more carefully, I was wrong, this doesn't work at all. I didn't notice there are two different queries you pass into `DetectionsTable` (candidates list vs single candidate), react query isn't normalized so you can only invalidate/update a specific query. It might just be best to go back to the way you had it originally, sorry…

Reply:
Just made a commit to undo the change, works fine now